isis/tests/TestCsmPlugin.cpp

```diff
@@ -0,0 +1,126 @@
+#include <iostream>
+
+#include "TestCsmPlugin.h"
+#include "TestCsmModel.h"
+
+// Static Instance of itself
+const TestCsmPlugin TestCsmPlugin::m_registeredPlugin;
+
+TestCsmPlugin::TestCsmPlugin() {
+  // deliberately blank
+}
+
+TestCsmPlugin::~TestCsmPlugin() {}
+
+std::string TestCsmPlugin::getPluginName() const {
+  return "TestCsmPlugin";
+}
+
+std::string TestCsmPlugin::getManufacturer() const {
+  return "TestCsmPluginCreator";
+}
+
+std::string TestCsmPlugin::getReleaseDate() const {
+  return "TestDate";
+}
+
+csm::Version TestCsmPlugin::getCsmVersion() const {
+  return csm::Version(3,0,3);
+}
+
+size_t TestCsmPlugin::getNumModels() const {
+  return 2;
+}
+
+std::string TestCsmPlugin::getModelName(size_t modelIndex) const {
+  return "TestModelName";
+}
+
+std::string TestCsmPlugin::getModelFamily(size_t modelIndex) const {
+  return "TestModelFamily";
+}
+
+csm::Version TestCsmPlugin::getModelVersion(const std::string& modelName) const { 
+  return csm::Version(3,0,3);
+}
+
+bool TestCsmPlugin::canModelBeConstructedFromState(const std::string& modelName,
+                                    const std::string& modelState,
+                                    csm::WarningList* warnings) const {
+  try {
+    csm::Model *model = constructModelFromState(modelState, warnings);
+    return static_cast<bool>(model);
+  }
+  catch (std::exception &e) {
+    std::string msg = "Could not create model [";
+    msg += modelName;
+    msg += "] with error [";
+    msg += e.what();
+    msg += "]";
+    std::cout << msg << std::endl;
+  }
```

Comment:
We shouldn't do this try/catch logic similar to what we have in usgscsm. I would instead check something in the ISD like the filename and then decide off that. So, if the filename matches "testisd" or something, then it can create the first model, if it matches "testisd2" then it can create the second model. This helps ensure that no tests accidentally use this plugin and model. Right now, this will always return True, which could be problematic.

Reply:
Yeah, this was one of the things that was "in progress" for modification to use the second model! I ended up doing something like this.

Reply:
Okay, that's what I figured 👍